streamer/pulp/streamer/server.py

```diff
@@ -232,6 +229,7 @@ def _download(self, request, entry, responder):
         finally:
             try:
                 downloader.config.finalize()
+                self.session_cache.put()
```

Comment:
I don't think this method exists, and that's not coming to your attention because of the lovely 2 lines below.